tasky/src/components/Header/Header.js

```diff
@@ -1,52 +1,73 @@
-import React from "react";
 import Image from "next/image";
-import { HeaderContainer } from "./styles";
-import { useRouter } from "next/router";
-const Header = () => {
-  const router = useRouter();
+import React, { useState } from "react";
+import { HeaderContainer, NavButton, Container } from "./styles";
+import Backlog from "@/pages/backlog";
+// import Calendar from "@/pages/calendar"; TODO: uncomment this line and implement the calendar page
+import Home from "@/pages/home";
+
+const NavHeader = (props) => {
+  const [activePage, setActivePage] = useState("/home"); // State to track active page
+
+  const handleNavigation = (page) => {
+    setActivePage(page);
+  };
+
   return (
-    <HeaderContainer>
-      <Image
-        width={160}
-        height={58}
-        src="./Logo.svg"
-        style={{ marginLeft: "15px" }}
-      />
-      <Image
-        width={131.5}
-        height={20}
-        src="./HomePageUC.svg"
-        style={{ marginLeft: "10%", marginTop: "2%", cursor: "pointer" }}
-        onClick={() => {
-          router.push("/homepage");
-        }}
-      />
-      <Image
-        width={108.5}
-        height={20}
-        src="./CalendarUC.svg"
-        style={{ marginLeft: "2%", marginTop: "2%", cursor: "pointer" }}
-        onClick={() => {
-          router.push("/calendar");
-        }}
-      />
-      <Image
-        width={99}
-        height={40}
-        src="./BacklogClicked.svg"
-        style={{ marginLeft: "2%", marginTop: "2%", cursor: "pointer" }}
-        onClick={() => {
-          router.push("/backlog");
-        }}
-      />
-      <Image
-        width={70}
-        height={38}
-        src="./ProfileClicked.svg"
-        style={{ marginLeft: "46%", marginTop: "1.2%", cursor: "pointer" }}
-      />
-    </HeaderContainer>
+    <Container>
+      <HeaderContainer>
+        <Image
+          width={160}
+          height={58}
+          src="./Logo.svg"
+          style={{ marginLeft: "15px" }}
+        />
+        <NavButton
+          active={activePage === "/home"}
+          onClick={() => handleNavigation("/home")}
+        >
+          Home
+        </NavButton>
+        <NavButton
+          active={activePage === "/calendar"}
+          onClick={() => handleNavigation("/calendar")}
+        >
+          Calendar
+        </NavButton>
+        <NavButton
+          active={activePage === "/backlog"}
+          onClick={() => handleNavigation("/backlog")}
+        >
+          Backlog
+        </NavButton>
+        <NavButton
+          active={activePage === "/profile"}
+          onClick={() => handleNavigation("/profile")}
+        >
+          Profile
+        </NavButton>
```

Comment:
Since all of these are the same component, its best practice to just have a list
`headerButtons = ['home', 'calendar', ....]`
`return (
  headerButtons.map((title)=>(
    <NavButton active={activatePage === `/${title.toLowercase()}` onClick={()=> handleNavigation(`/${title.toLowercase()}`)}>
    title
    </NavButton>
  )
)`

Reply:
Are you talking only on the navButtons declarations?